Guess the right name for COMPOSE_PROJECT_NAME

mstriemer opened the pull request.

muffinresearch commented on docs/topics/install/docker.rst:
```diff
@@ -103,6 +103,13 @@ on each container. Here is an example of running the ``py.test`` command on the
+.. note::
+    If you see an error like ``No such container: addonsserver_web_1`` please
+    ensure that your copy of addons-server is in a directory named
+    ``addons-server``. If you had an old checkout that was called ``olympia``
```
Or you can set the env var

mstriemer replied:
Totally, I'll add that.

mstriemer commented:
I updated the `Makefile` to detect the name of the directory that you're in which is what `docker-compose` defaults to.

I also updated the documentation to reflect what to do if that's somehow not what the containers are called but hopefully this stops happening.

tofumatt commented on docs/index.rst:
```diff
@@ -15,5 +15,5 @@ Contents
 
    topics/api/index
-   topics/install-olympia/index
+   topics/install/index
```
You'll want to update the README's link to install docs to make sure the README doesn't break.

mstriemer replied:
Thanks, updated.

mstriemer commented:
This actually doesn't work since `docker-compose` strips (at least) `-`s.

mstriemer commented:
This strips `-`s and `_`s now. Hopefully that's all people have in their folder names. If you have a space in the name then it won't work. I can't imagine someone doing that, but who knows.

kumar303 commented on Makefile:
```diff
@@ -2,7 +2,11 @@
-COMPOSE_PROJECT_NAME?=addonsserver
+# Get the name of the Makefile's directory for the docker container base name.
+mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+current_dir := $(notdir $(patsubst %/,%,$(dir $(mkfile_path))))
```
![fail-kids](https://cloud.githubusercontent.com/assets/55398/13021269/e4a6e612-d19f-11e5-9dab-881384a5c831.gif)

because Makefile

kumar303 replied:
(not a comment suggesting you should change anything, just me trying to read the code is all)

mstriemer replied:
Definitely copy/pasted from http://stackoverflow.com/questions/18136918/how-to-get-current-directory-of-your-makefile.

kumar303 replied:
but while I'm at it, it would be pretty nice to rewrite this as a fabric script or something

kumar303 commented on Makefile:
```diff
@@ -2,7 +2,11 @@
+mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+current_dir := $(notdir $(patsubst %/,%,$(dir $(mkfile_path))))
+
+COMPOSE_PROJECT_NAME?=$(shell echo "${current_dir}" | tr -d '-' | tr -d '_')
```
I hope this works all of the time

kumar303 commented on README.rst:
```diff
@@ -19,5 +19,5 @@ You can access the AMO stage environment at https://addons.allizom.org/
 
 .. _`addons.mozilla.org`: https://addons.mozilla.org
-.. _`install docs`: https://addons-server.readthedocs.org/en/latest/topics/install-olympia/docker.html
+.. _`install docs`: https://addons-server.readthedocs.org/en/latest/topics/install/docker.html
```
This should be a ref instead. Change the link to ``:ref:`install-with-docker```. This is a ref that's already defined. You can see the definition in `docker.rst` as `.. _install-with-docker:`.

mstriemer replied:
This is in the README though so I don't think that will work? I don't think GitHub would know to link to https://addons-server.readthedocs.org so I don't know where the link would go.

kumar303 replied:
whoops, ignore me. I mis-read the diff.

kumar303 commented:
I don't think this is so necessary because new contributors will most likely always run `git clone ...` which by default will put the code in a directory called `addons-server`. But I suppose it wouldn't hurt to account for someone who clones to a custom directory.

r+wc
